packages/node/src/handlers.ts

```diff
@@ -32,19 +32,42 @@ export function tracingHandler(): (
     const reqMethod = (req.method || '').toUpperCase();
     const reqUrl = req.url;
 
-    const hub = getCurrentHub();
-    const transaction = hub.startSpan({
+    let traceId;
+    let parentSpanId;
+
+    // If there is a trace header set, we extract the data from it and set the span on the scope
+    // to be the origin an created transaction set the parent_span_id / trace_id
+    if (req.headers && isString(req.headers['sentry-trace'])) {
+      const span = Span.fromTraceparent(req.headers['sentry-trace'] as string);
+      if (span) {
+        const spanData = span.toJSON();
+        traceId = spanData.trace_id;
+        parentSpanId = spanData.span_id;
+      }
+    }
+
+    const transaction = startTransaction({
+      name: `${reqMethod} ${reqUrl}`,
       op: 'http.server',
-      transaction: `${reqMethod} ${reqUrl}`,
+      parentSpanId,
+      traceId,
     });
 
-    hub.configureScope(scope => {
-      scope.setSpan(transaction);
-    });
+    if (transaction) {
+      // We put the transaction on the scope so users can attach children to it
+      getCurrentHub().configureScope(scope => {
+        scope.setSpan(transaction);
+      });
+      // We also set a function getTransaction on the response so people can grab the transaction there to add
+      // spans to it
+      (res as any).getTransaction = () => transaction;
```

Comment:
I wonder if there's a way to type it somehow, so people can see autocompletion in TS code.

Reply:
I changed this to just a variable, then this ugly assignment `const transaction = (res as any).getTransaction && (res as any).getTransaction();` becomes 5% less ugly
`const transaction = (res as any).__sentry_transaction;`